src/runtime/recipe/recipe-util.ts

```diff
@@ -394,7 +394,7 @@ export class RecipeUtil {
         newMatches = newMatches.concat(
             RecipeUtil._assignHandlesToEmptyPosition(shape, match, [thisHandle], nullHandles));
       } else {
-        newMatches.concat(match);
+        newMatches = newMatches.concat(match);
```

Comment:
This looks like it found a bug to me, maria?

Reply:
not my code, but definitely looks like you did. 
@shans @Cypher1 - do we want to add a unit test for this?

Reply:
We should definitely test this and figure out if it created a hidden bug.